pkg/operator/controller/garden/reconciler_delete.go

```diff
@@ -93,7 +93,11 @@ func (r *Reconciler) delete(
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	kubeAPIServerService := r.newKubeAPIServerService(log, garden)
+	if len(istio.GetValues().IngressGateway) != 1 {
+		return reconcile.Result{}, fmt.Errorf("exactly one Istio Ingress Gateway is required")
+	}
+	kubeAPIServerSNI := r.newSNI(garden, istio.GetValues().IngressGateway[0])
```

Comment:
Perhaps it's better to perform this check in the `newSNI` function instead of duplicating it in both instantiations? Then you can pass all ingress gateways to `newSNI` and let it pick the first one after the check succeeded.